main.js

```diff
@@ -0,0 +1,802 @@
+      alt1: '<',
+      alt2: null,
+      code: 'Comma',
+    },
+    {
+      value1: '.',
+      value2: 'ю',
+      alt1: '>',
+      alt2: null,
+      code: 'Period',
+    },
+    {
+      value1: '/',
+      value2: '.',
+      alt1: '?',
+      alt2: ',',
+      code: 'Slash',
+    },
+    {
+      value1: '',
+      value2: '',
+      alt1: null,
+      alt2: null,
+      code: 'ArrowUp',
+    },
+    {
+      value1: '',
+      value2: '',
+      alt1: null,
+      alt2: null,
+      code: 'ShiftRight',
+    },
+  ],
+  [
+    {
+      value1: '',
+      value2: '',
+      alt1: null,
+      alt2: null,
+      code: 'ControlLeft',
+    },
+    {
+      value1: '',
+      value2: '',
+      alt1: null,
+      alt2: null,
+      code: 'MetaLeft',
+    },
+    {
+      value1: '',
+      value2: '',
+      alt1: null,
+      alt2: null,
+      code: 'AltLeft',
+    },
+    {
+      value1: ' ',
+      value2: ' ',
+      alt1: null,
+      alt2: null,
+      code: 'Space',
+    },
+    {
+      value1: '',
+      value2: '',
+      alt1: null,
+      alt2: null,
+      code: 'AltRight',
+    },
+    {
+      value1: '',
+      value2: '',
+      alt1: null,
+      alt2: null,
+      code: 'ControlRight',
+    },
+    {
+      value1: '',
+      value2: '',
+      alt1: null,
+      alt2: null,
+      code: 'ArrowLeft',
+    },
+    {
+      value1: '',
+      value2: '',
+      alt1: null,
+      alt2: null,
+      code: 'ArrowDown',
+    },
+    {
+      value1: '',
+      value2: '',
+      alt1: null,
+      alt2: null,
+      code: 'ArrowRight',
+    },
+  ],
+];
+
+const keyboard = {
+  elements: {
+    main: null,
+    keys: [],
+  },
+
+  properties: {
+    value: '',
+    capslock: false,
+    shift: false,
+    lang: localStorage.getItem('lang') || 'eng',
+    lastCaretPos: 0,
+  },
+
+  initilize() {
+    localStorage.setItem('isRus', this.properties.isRus);
+    this.elements.main = document.createElement('div');
+    this.elements.main.classList.add('keyboard');
+    this.elements.main.appendChild(this.createKeys());
+
+    this.elements.keys = this.elements.main.querySelectorAll('.key');
+    document.body.appendChild(this.elements.main);
+  },
+
+  createKeys() {
+    const fragment = document.createDocumentFragment();
+
+    keysLayout.forEach((line) => {
+      const row = document.createElement('div');
+      row.classList.add('keyboard__row');
+
+      line.forEach((key) => {
+        const keyElement = document.createElement('button');
+        keyElement.setAttribute('type', 'button');
+        keyElement.setAttribute('data-key', `${key.code}`);
+        keyElement.classList.add('key');
+
+        switch (key.code) {
+          case 'Backspace':
+            keyElement.classList.add('backspace', 'dark');
+            keyElement.innerHTML = '<i class="fas fa-backspace"></i>';
+            keyElement.addEventListener('click', () => {
+              textarea.value = textarea.value.substring(0, textarea.value.length - 1);
```

Comment:
Better use this method: https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/String/slice.
Then it will look like that: textarea.value.slice(0, -1);